### PR TITLE
doc: security: Add hardening tool information

### PR DESCRIPTION
**Pull request by ceolin**
Add basic reference to hardening tool.

Signed-off-by: Flavio Ceolin <flavio.ceolin@intel.com>

### CODE REVIEW

**Review by d3zd3z — APPROVED** [review]

File: doc/security/hardening-tool.rst [hunk 1]
```diff
@@ -0,0 +1,44 @@
+.. _hardening:
+
+Hardening Tool
+##############
+
+Zephyr contains several options features that makes the overall system
```
Comment (nashif) [inline doc/security/hardening-tool.rst]: optional

File: doc/security/hardening-tool.rst [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+more secure. As we take advantage of hardware features, many of these
+options are platform specific and besides it, some of them are not
+known by developers.
+
+To address this problem, we provide a tool that helps to check an
+application Kconfig option list against a list of hardening
```
Comment (nashif) [inline doc/security/hardening-tool.rst]: application configuration option

File: doc/security/hardening-tool.rst [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+preferences defined by the **Security Group**. The tool is capable of
+understand what is the build target and suggest specific
```
Comment (nashif) [inline doc/security/hardening-tool.rst]: understanding

File: doc/security/hardening-tool.rst [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+After configure your application, cd to the build folder and:
+
+.. code-block:: console
+
+   # ninja build system:
+   $ ninja hardenconfig
+   # make build system:
+   $ make hardenconfig
+
+The output of should be similar to the one bellow:
```
Comment (nashif) [inline doc/security/hardening-tool.rst]: remove of

File: doc/security/hardening-tool.rst [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+To address this problem, we provide a tool that helps to check an
+application Kconfig option list against a list of hardening
+preferences defined by the **Security Group**. The tool is capable of
+understand what is the build target and suggest specific
+recommendations to it.
+
+Usage
+*****
+
+After configure your application, cd to the build folder and:
```
Comment (nashif) [inline doc/security/hardening-tool.rst]: after configuration of your application, change directory to the build folder...

**Review by nashif — CHANGES_REQUESTED** [review]

File: doc/security/hardening-tool.rst [hunk 6]
```diff
@@ -0,0 +1,44 @@
+.. _hardening:
+
+Hardening Tool
+##############
+
+Zephyr contains several optional features that makes the overall system
```
Comment (nashif) [inline doc/security/hardening-tool.rst]: that make....

File: doc/security/hardening-tool.rst [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+more secure. As we take advantage of hardware features, many of these
+options are platform specific and besides it, some of them are not
+known by developers.
```
Comment (nashif) [inline doc/security/hardening-tool.rst]: not known = unknown

File: doc/security/hardening-tool.rst [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+
+To address this problem, we provide a tool that helps to check an
```
Comment (nashif) [inline doc/security/hardening-tool.rst]: s/we/Zephyr/

File: doc/security/hardening-tool.rst [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+application configuration option list against a list of hardening
+preferences defined by the **Security Group**. The tool is capable of
+understanding what is the build target and suggest specific
```
Comment (nashif) [inline doc/security/hardening-tool.rst]: The tool can identify the build target and based on that provides suggestions and recommendations on how to optimize the configuration for security..

**Review by nashif — COMMENTED** [review]
a few things more...

**Review by nashif — APPROVED** [review]
a few things more...

**Pull request merged**